projects/rift-finance/index.js

```diff
@@ -0,0 +1,166 @@
+const sdk = require("@defillama/sdk");
+
+const { getBlock } = require("../helper/getBlock");
+const { unwrapUniswapLPs } = require("../helper/unwrapLPs");
+const { getChainTransform } = require("../helper/portedTokens");
+
+const ADDRESSES = require("./addresses");
+
+const riftVaultAbi = require("./abis/riftVaultAbi");
+const masterChefAbi = require("./abis/masterChefAbi");
+
+function getAbi(obj, fnName) {
+  return obj.find((x) => x.name === fnName);
+}
+
+async function getChainBalances(timestamp, chainBlocks, chain) {
+  const block = await getBlock(timestamp, chain, chainBlocks);
+  const balances = {};
+  const transform = await getChainTransform(chain);
+
+  const { core, startBlock } = ADDRESSES[chain];
+
+  // Get all registered vaults by searching events on core address.
+  const vaultRegisteredEvents = (
+    await sdk.api.util.getLogs({
```

Comment:
Sorry, is there some other way to get the vaults? we do not have a full node in aurora and this query tends get slower over time and would start failing

Reply:
Yeah, it might work in ethereum but not in other chains :( we dont have ful node rpc for this, can you try setting up a graph api for this?

Reply:
We've removed this in favor of a gql query to our subgraph in order to retrieve the registered vaults.